src/main/kotlin/no/nav/familie/ef/sak/selvstendig/NæringsinntektKontrollService.kt

```diff
@@ -0,0 +1,132 @@
+
+            oppgaver.forEach { oppgave ->
+                val personIdent =
+                    oppgave.identer?.firstOrNull { it.gruppe == IdentGruppe.FOLKEREGISTERIDENT }?.ident
+                        ?: oppgave.personident ?: throw Exception("Fant ikke registrert ident på oppgave ${oppgave.id}")
+
+                secureLogger.info("Kontrollerer person med ident: $personIdent")
+
+                val fagsakOS = fagsakService.finnFagsaker(setOf(personIdent)).filter { it.stønadstype == StønadType.OVERGANGSSTØNAD }.firstOrNull() ?: throw RuntimeException("Fant ikke fagsak for overgangsstønad for person: $personIdent")
+                val behandling = behandlingService.finnSisteIverksatteBehandling(fagsakOS.id) ?: throw RuntimeException("Fant ikke behandling for fagsakId: ${fagsakOS.id}")
+                val antallMånederMedVedtakForFjoråret = antallMånederMedVedtakForFjoråret(behandling)
+
+                val næringsinntekt = hentFjoråretsNæringsinntekt(fagsakOS.fagsakPersonId)
+
+                if (antallMånederMedVedtakForFjoråret > 3 && næringsinntekt > 50_000) {
+                    val fjoråretsPersoninntekt = inntektService.hentÅrsinntekt(personIdent, årstallIFjor)
+                    secureLogger.info("Forrige års inntekt for person uten ytelse fra offentlig: $fjoråretsPersoninntekt")
+                    if (fjoråretsPersoninntekt == 0) {
+                        val forventetInntekt = forventetInntektSnitt(behandling.id)
+                        secureLogger.info("Beregnet inntekt i snitt for år $årstallIFjor og behandlingId ${behandling.id} er: $forventetInntekt")
+                        if (næringsinntekt > (forventetInntekt * 1.1)) {
+                            secureLogger.info("Har 10% høyere næringsinntekt for person: $personIdent (Næringsinntekt: $næringsinntekt - ForventetInntekt: $forventetInntekt)")
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private fun hentFjoråretsNæringsinntekt(fagsakPersonId: UUID): Int {
+        val inntekt = sigrunService.hentInntektForAlleÅrMedInntekt(fagsakPersonId).filter { it.inntektsår == årstallIFjor }
+
+        val næringsinntekt = inntekt.sumOf { it.næring } + inntekt.sumOf { it.svalbard?.næring ?: 0 }
+        secureLogger.info("Inntekt for person $inntekt - næringsinntekt er beregnet til: $næringsinntekt")
+        return næringsinntekt
+    }
+
+    private fun forventetInntektSnitt(behandlingId: UUID): Int {
+        val vedtak = vedtakService.hentVedtak(behandlingId)
+
+        val inntektsperioderIÅr = vedtak.inntekter?.inntekter?.filter { it.periode.overlapper(Månedsperiode(YearMonth.of(årstallIFjor, 1), YearMonth.of(årstallIFjor, 12))) }
+        val totalInntekt = inntektsperioderIÅr?.sumOf { it.totalinntekt() }
+
+        val antallPerioder = inntektsperioderIÅr?.size ?: throw RuntimeException("Fant ikke inntektsperiode på vedtak for år $årstallIFjor for behandling $behandlingId")
```

Comment:
Henger ikke helt med her, men er det gitt at alle vi itererer igjennom hadde vedtak for fjoråret? Denne smellen vil vel krasje alle sjekkene?